qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -151,9 +156,26 @@ def backends(
         backends: List[IBMBackend] = []
         if instance:
             hgp = self._provider._get_hgp(instance=instance)
-            backends = list(hgp.backends.values())
+            for backend_name in hgp.backends.keys():
+                if (
+                    not self._backends[backend_name]
+                    or instance != self._backends[backend_name]._instance
+                ):
+                    self._backends[backend_name] = self._fetch_backend_config(
+                        backend_name, instance
+                    )
```

Comment:
Instead of fetching the same backend config for each hgp, perhaps we can store the backend configs in a local variable and dynamically construct the `IBMBackend` instances. For example, something like

```
if not self._backends_config[backend_name]:
    self._backend_config[backend_name] = self._fetch_backend_config(backend_name)
backends.append(IBMBackend(instance, self._backend_config[backend_name]))
````

this would be a lot of cheaper than making all server requests. Downside is if one changes the config of one `IBMBackend` instance, it'd also affect another instance of the same backend (but different hgp). But people really shouldn't be changing backend configs lol

Reply:
added in [2679d4a](https://github.com/Qiskit/qiskit-ibm-provider/pull/545/commits/2679d4ab6b107b851a5d59ce17f46b594e59bbec), seems to be noticeably faster now